client/src/components/Doctor/ManageMedicalrecords.tsx

```diff
@@ -0,0 +1,441 @@
+import React, { FunctionComponent, useContext, useState, useEffect } from 'react';
+import ReactPaginate from 'react-paginate';
+import { HealthContext } from '../../providers/HealthProvider';
+import AWS from 'aws-sdk';
+import axios from 'axios';
+import { AuthContext } from '../../providers/AuthProvider';
+import { useAuthFetch } from '../../hooks/api';
+
+interface AppointmentDetails {
+  doctor_email: string;
+  patient_email: string;
+  appointment_id: string;
+  patient_name: string;
+  doctor_address: string;
+  patient_address: string;
+  created_at: string;
+  appointment_time: string;
+  appointment_status: string;
+}
+
+const ManageMedicalRecords: FunctionComponent<{}> = () => {
+  const { user, role, logout } = useContext(AuthContext);
+  const { fetch } = useAuthFetch();
+  const [confirmedAppointmentData, setConfirmedAppointmentData] = useState<AppointmentDetails[]>(
+    [],
+  );
+  const { requestMedicalRecordHealthBlockContract } = useContext(HealthContext);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [filterOption, setFilterOption] = useState<string>('all');
+  const [showRequestPopup, setShowRequestPopup] = useState(false);
+  const [selectedFile, setSelectedFile] = useState<undefined | File>(undefined);
+  const [showUploadPopup, setShowUploadPopup] = useState(false);
+
+  useEffect(() => {
+    getConfirmedAppointment();
+  }, []);
+  const getConfirmedAppointment = async () => {
+    let resp = await fetch('GET', '/confirmedAppointments/' + user?.email);
+    if (resp && resp.data && resp.data.result) {
+      setConfirmedAppointmentData(resp.data.result);
+    } else {
+      setConfirmedAppointmentData([]);
+    }
+  };
+
+  const handleMedicalRecordRequest = async (
+    patientAddress: string,
+    patientName: string,
+    docEmail: string,
+    patientEmail: string,
+  ) => {
+    try {
+      await requestMedicalRecordHealthBlockContract?.(
+        patientAddress,
+        docEmail,
+        patientName,
+        patientEmail,
+      );
+      alert('Medical record request sent successfully');
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  // function sendRequestEmail(to: string) {
```

Comment:
remove commented code if it is not necessary

Reply:
That code was working but that is in frontend so I commented for now. Shruthi is fixing that ses service to go backend. Will remove this after I get shruthi's changes.